src/controllers/dynakube/activegate/internal/capability/reconciler.go

```diff
@@ -2,165 +2,57 @@ package capability
+		dynakube:                   dynakube,
+		client:                     clt,
 	}
-	return nil, errors.Errorf(`Cannot find container "%s" in the provided slice (len %d)`,
-		containerName, len(containers),
-	)
 }
 
-func getActiveGateContainer(sts *appsv1.StatefulSet) (*corev1.Container, error) {
-	return getContainerByName(sts.Spec.Template.Spec.Containers, consts.ActiveGateContainerName)
-}
-
-func setCommunicationsPort(dk *dynatracev1beta1.DynaKube) kubeobjects.StatefulSetEvent {
-	return func(sts *appsv1.StatefulSet) {
-		activeGateContainer, err := getActiveGateContainer(sts)
-		if err == nil {
-			activeGateContainer.Ports = []corev1.ContainerPort{
-				{
-					Name:          consts.HttpsServicePortName,
-					ContainerPort: httpsContainerPort,
-				},
-				{
-					Name:          consts.HttpServicePortName,
-					ContainerPort: httpContainerPort,
-				},
-			}
-		} else {
-			log.Info("Cannot find container in the StatefulSet", "container name", consts.ActiveGateContainerName)
-		}
-
-		if dk.NeedsStatsd() {
-			statsdContainer, err := getContainerByName(sts.Spec.Template.Spec.Containers, consts.StatsdContainerName)
-			if err == nil {
-				statsdContainer.Ports = []corev1.ContainerPort{
-					{
-						Name:          consts.StatsdIngestTargetPort,
-						ContainerPort: consts.StatsdIngestPort,
-						Protocol:      corev1.ProtocolUDP,
-					},
-				}
-			} else {
-				log.Info("Cannot find container in the StatefulSet", "container name", consts.StatsdContainerName)
-			}
-		}
-	}
-}
-
-func (r *Reconciler) calculateStatefulSetName() string {
-	return capability.CalculateStatefulSetName(r.Capability, r.Dynakube.Name)
-}
-
-func addDNSEntryPoint(dynakube *dynatracev1beta1.DynaKube, moduleName string) kubeobjects.StatefulSetEvent {
-	return func(sts *appsv1.StatefulSet) {
-		if activeGateContainer, err := getActiveGateContainer(sts); err == nil {
-			activeGateContainer.Env = append(activeGateContainer.Env,
-				corev1.EnvVar{
-					Name:  dtDnsEntryPoint,
-					Value: buildDNSEntryPoint(dynakube, moduleName),
-				})
-		} else {
-			log.Error(err, "Cannot find container in the StatefulSet", "container name", consts.ActiveGateContainerName)
-		}
-	}
-}
-
-func buildDNSEntryPoint(dynakube *dynatracev1beta1.DynaKube, moduleName string) string {
-	return fmt.Sprintf("https://%s/communication", BuildServiceHostName(dynakube.Name, moduleName))
-}
+type NewReconcilerFunc = func(clt client.Client, capability capability.Capability, dynakube *dynatracev1beta1.DynaKube, statefulsetReconciler kubeobjects.Reconciler, customPropertiesReconciler kubeobjects.Reconciler) *Reconciler
 
 func (r *Reconciler) Reconcile() (update bool, err error) {
 	_, err = r.customPropertiesReconciler.Reconcile()
 	if err != nil {
 		return update, errors.WithStack(err)
 	}
 
-	if r.ShouldCreateService() {
+	if r.capability.ShouldCreateService() {
 		// TODO: MutliCapability shouldn't be used here - it may be as well one of deprecated Capabilities: Kubemon or Routing
```

Comment:
Maybe delete the TODO, it exists since for forever, so it might as well be deleted